### PR TITLE
Convert all `try!` to use `?` operator

### PR DESCRIPTION
**Pull request by Kintaro**
As there are no guidelines on the usage of `try!` vs `?`, this pull request changes all occurences to `?`.
Mostly done automated via `rustfmt`.

Makes the code more idomatic and less cluttered with parentheses.

If the `try!` is part of the coding guidelines, this PR can be ignored and closed.

### CODE REVIEW

File: src/engine/strat_engine/thinpool.rs [hunk 1]
```diff
@@ -166,14 +166,14 @@ impl ThinPool {
...
-        try!(self.mdv.check());
+        let thinpool = self.thin_pool.status(dm)?;
+        self.mdv.check()?;
 
-        let filesystems: Vec<_> = try!(self.filesystems
-                                           .borrow_mut()
-                                           .into_iter()
-                                           .map(|fs| fs.check(dm))
-                                           .collect());
+        let filesystems = self.filesystems
```
Comment (mulkieran) [inline src/engine/strat_engine/thinpool.rs]: If you moved the type annotation from after ```collect``` to after ```filesystems``` would it still compile?

Comment (Kintaro) [inline src/engine/strat_engine/thinpool.rs]: That, and mostly the preferred way is to put it after collect. It allows collect to do a bit of magic here and there.

**Comment by mulkieran** [comment]
I like everything except for the final "Merge" commit. @agrover What do you think?

**Comment by Kintaro** [comment]
I can squash and rebase as soon as I get home

**Comment by agrover** [comment]
👍

**Comment by Kintaro** [comment]
Squashed and rebased. Merge commit is now gone.

File: src/engine/macros.rs [hunk 2]
```diff
@@ -29,9 +29,9 @@ macro_rules! destroy_pool {
         } else {
             return Ok(false);
         }
-        try!($s.pools.remove_by_uuid($uuid)
-             .expect("Must succeed since $s.pool.get_by_uuid() returned a value")
-             .destroy());
+        $s.pools.remove_by_uuid($uuid)
+             .expect("Must succeed since $s.pool.get_by_uuid() returned a value.")
```
Comment (mulkieran) [inline src/engine/macros.rs]: Whoops! This added a '.' back in.

Comment (Kintaro) [inline src/engine/macros.rs]: Sorry about that. Fixed.

**Comment by mulkieran** [comment]
@agrover It all looks good to me, and I believe the tests should pass this time.

**Comment by agrover** [comment]
All right, let's do it.

**Pull request merged**